### PR TITLE
[4.0] Move back-to-top to custom module

### PR DESCRIPTION
**Pull request by C-Lodder**
Pull Request for Issue #25239 

### Summary of Changes

The "Back to Top" button is now a custom module. This allows users to disable it if need be.
The logic has also changed so that even if you have disabled the footer module but have enabled the `back-top` module, it will still display.

### Testing Instructions

1. Install sample data
2. Ensure the "Back to Top" button is enabled
-----------------------------------
1. Disable the footer module
2. Ensure the "Back to Top" button is enabled

### CODE REVIEW

**Comment by infograf768** [comment]
As the old (broken) code is now totally out of cassiopea index.php, would not it be sensible to make it a non-custom specific module ? I understand it would mean also a separate css not specific to cassiopea.
I am afraid otherwise that as most install will not use testing sample data, this would  be a new hidden feature.

**Comment by C-Lodder** [comment]
@infograf768 - Fair point. I really don't mind either way. Will let the powers that be decide that.

**Comment by infograf768** [comment]
Note: also see this RTC https://github.com/joomla/joomla-cms/pull/23661

**Comment by C-Lodder** [comment]
@infograf768 That's simply using CSS-Grid for the sidebars....not related to this

File: installation/sql/mysql/sample_testing.sql [hunk 1]
```diff
@@ -651,7 +652,8 @@ INSERT INTO `#__modules` (`id`, `title`, `note`, `content`, `ordering`, `positio
...
 (120, 'Footer', '', '&copy; Joomla! CMS', 1, 'footer', 0, NULL, NULL, NULL, 1, 'mod_custom', 1, 1, '', 0, '*'),
-(121, 'Latest Actions', '', '', 0, 'cpanel', 0, '0000-00-00 00:00:00', '0000-00-00 00:00:00', '0000-00-00 00:00:00', 1, 'mod_latestactions', 6, 1, '{}', 1, '*');
+(121, 'Latest Actions', '', '', 0, 'cpanel', 0, '0000-00-00 00:00:00', '0000-00-00 00:00:00', '0000-00-00 00:00:00', 1, 'mod_latestactions', 6, 1, '{}', 1, '*'),
+(122, 'Back to Top', '', '<a href=\"#top\" id=\"back-top\" class=\"back-top\">\r\n<span class=\"icon-arrow-up-4\" aria-hidden=\"true\"></span>\r\n<span class=\"sr-only\">Back to Top</span>\r\n</a>', 1, 'back-top', 524, '2019-06-17 07:35:31', NULL, NULL, 1, 'mod_custom', 1, 1, '{\"prepare_content\":1,\"backgroundimage\":\"\",\"layout\":\"_:default\",\"moduleclass_sfx\":\"\",\"cache\":1,\"cache_time\":900,\"cachemode\":\"static\",\"style\":\"0\",\"module_tag\":\"div\",\"bootstrap_size\":\"0\",\"header_tag\":\"h3\",\"header_class\":\"\"}', 0, '*');
```
Comment (Quy) [inline installation/sql/mysql/sample_testing.sql:656]: No need to escape `”`. See line 652.

File: templates/cassiopeia/scss/blocks/_css-grid.scss [hunk 2]
```diff
@@ -1,103 +1,100 @@
...
+
+  > .full-width {
+    grid-column: full-start / full-end;
+  }
+
+  @include media-breakpoint-down(xs) {
     grid-template-areas:
       ". head head head head ."
       ". banner banner banner banner ."
+      ". main main main main ."
```
Comment (wilsonge) [inline templates/cassiopeia/scss/blocks/_css-grid.scss:49]: Moving main above top positions seems wrong to me

File: installation/sql/mysql/sample_testing.sql [hunk 3]
```diff
@@ -208,7 +208,8 @@ INSERT INTO `#__assets` (`id`, `parent_id`, `lft`, `rgt`, `level`, `name`, `titl
 (212, 18, 363, 364, 2, 'com_modules.module.119', 'Banner', '{"core.delete":[],"core.edit":[],"core.edit.state":[],"module.edit.frontend":[]}'),
 (213, 18, 365, 366, 2, 'com_modules.module.120', 'Footer', '{"core.delete":[],"core.edit":[],"core.edit.state":[],"module.edit.frontend":[]}'),
 (214, 1, 367, 368, 1, 'com_privacy', 'com_privacy', '{"core.admin":{"7":1}}'),
-(215, 1, 369, 370, 1, 'com_actionlogs', 'com_actionlogs', '{"core.admin":{"7":1}}');
+(215, 1, 369, 370, 1, 'com_actionlogs', 'com_actionlogs', '{"core.admin":{"7":1}}'),
+(216, 18, 371, 372, 2, 'com_modules.module.122', 'Back to Top', '{"core.delete":[],"core.edit":[],"core.edit.state":[],"module.edit.frontend":[]}');
```
Comment (wilsonge) [inline installation/sql/mysql/sample_testing.sql:212]: These assets are now wrong. Last time i used https://gist.github.com/alexgarel/b4e4a2e3160d3b0e8559 to correct (can help you with using this if you ping me) or any other joomla asset fixer script works too

Comment (brianteeman) [reply installation/sql/mysql/sample_testing.sql:212]: is there any point even keeping that file ?

File: templates/cassiopeia/scss/blocks/_css-grid.scss (hunk 2 above)
Comment (C-Lodder) [reply templates/cassiopeia/scss/blocks/_css-grid.scss:49]: Oh that must have been a mistake

File: installation/sql/mysql/sample_testing.sql (hunk 3 above)
Comment (C-Lodder) [reply installation/sql/mysql/sample_testing.sql:212]: @wilsonge Why do we need to run a 3rd party script to fix the assets?

Comment (brianteeman) [reply installation/sql/mysql/sample_testing.sql:212]: We don't have our own :(
It should be part of the core imho

Comment (C-Lodder) [reply installation/sql/mysql/sample_testing.sql:212]: Tragic

**Pull request closed without merging**